tuf/api/metadata.py

```diff
@@ -653,30 +652,63 @@ class VerificationResult:
     """Signature verification result for delegated role metadata.
 
     Attributes:
-        verified: True, if threshold of signatures is met.
-        signed: Set of delegated keyids, which validly signed.
-        unsigned: Set of delegated keyids, which did not validly sign.
+        threshold: Number of required signatures.
+        signed: dict of keyid:Key containing the keys that have signed.
+        unsigned: dict of keyid:Key containing the keys that have not signed.
+    """
+
+    threshold: int
+    signed: Dict[str, Key]
+    unsigned: Dict[str, Key]
+
+    def __bool__(self) -> bool:
+        return self.verified
+
+    @property
+    def verified(self) -> bool:
+        """True if threshold of signatures is met."""
+        return len(self.signed) >= self.threshold
+
+
+@dataclass
+class RootVerificationResult:
+    """Signature verification result for root metadata.
+
+    Root must be verified by itself and the previous root version. This
+    dataclass represents that combination. For the edge case of first version
+    of root, the same VerificationResult can be used twice.
```

Comment:
I had to read `get_root_verification_result` to understand what this means. Maybe rephrase to something like:


```suggestion
    dataclass represents both results. For the edge case of first version
    of root, the results are identical.
```